pgmuvi/lightcurve.py

```diff
@@ -328,6 +328,41 @@ def __init__(self, xdata, ydata, yerr=None,
         self.__FITTED_MAP = False
         self.__FITTED_MCMC = False
 
+    @classmethod
+    def from_table(cls, VOTable, **kwargs):
+        """Instantiate a Lightcurve object with
+        data read in from a VOTable.
+
+        Parameters
+        ----------
+        VOTable : str
+            Name (with extension) of the VOTable containing input data
+        kwargs:
+            Arguments to be passed to the Lightcurve call
+        
+        Returns
+        ----------
+        Lightcurve object
+        """
+        from astropy.table import Table
+        data = Table.read(VOTable, format='votable')
+        c = data.colnames
+        if ('x' not in c) or ('y' not in c):
+            raise ValueError('Table must have columns x, y, and yerr')
+        ndim = data['x'].squeeze().shape
```

Comment:
I;m guessing you've tested it and this does what you want, but I'm a bit confused as to how - squeeze() removed len(1) dimensions, so twon't this return a tuple of (n_obs,) or (2, n_obs,), which is a little confusing. I realise that you're checking it's len() later, but why not do that here? If there's a reason, please add more comments.

Reply:
No, you're right, the `len()` bit didn't transfer properly to this line. I'll fix it.

Reply:
Fixed.